### DISCUSSION TITLE
Twitter OAuth {"error":"requested path is invalid"}

### DISCUSSION THREAD

**Discussion by jonathanbouwens** [supabase/supabase#29510] [Questions]
Hi!

I added the Twitter login option to my web app, but it does not seem to work. As far as I know I have setup everything correctly;

- Enabled Twitter in providers
- Added callback URL (https://sgrqfnypkbooqbfewkja.supabase.co/auth/v1/callback) to Twitter developer
- Added API key to superbase Twitter provided

However, when I try to login using the Twitter function I am redirected to this page: https://sgrqfnypkbooqbfewkja.supabase.co/ with the message {"error":"requested path is invalid"}. Has anyone seen this before?

![image](https://github.com/user-attachments/assets/97ad110b-e592-4e87-a34e-6c5bbbe24efe)

<img width="997" alt="image" src="https://github.com/user-attachments/assets/e9729087-828b-4f53-bf57-d81b1ed62574">

<img width="998" alt="image" src="https://github.com/user-attachments/assets/1a8e0c2a-1b8b-4b82-9569-d39c4b84c554">

So far I haven't been able to fix this and I would love your help!

Cheers
Jonathan

**Comment by jonathanbouwens** [comment]
solved

**Reply by Marc4k** [reply]
Hey!
I have the same error how did you solve it?

**Reply by jonathanbouwens** [reply]
Hi! I copied the wrong api keys from Twitter developer. Not sure if that is the same case for you, but after copying the consumer keys instead of the OAuth keys it worked for me.

**Comment by letcontactabhi** [comment]
i have same issue but fixed copied the wrong api keys from Twitter developer

**Comment by weansard** [comment]
SOLVED:

You must use the API KEY and API SECRET KEY from X developer dashboard rather than using the keys labeled as OAuth keys.

**Reply by 0xVaibhav11** [reply]
They should update their docs! as the don't support OAuth 2.0
### 

<img width="659" alt="Screenshot 2025-02-14 at 5 16 23 AM" src="https://github.com/user-attachments/assets/0bb40d74-722a-4b3d-a62b-dd88f81dd809" />

**Comment by fingoldo** [comment]
supabase support, PLEASE update your user guide. only CONSUMER X keys work here.
